ci/build.ps1

```diff
@@ -35,61 +43,83 @@ if ($compiler -eq 'MINGW') {
     $nvimCmakeVars['USE_GCOV'] = 'ON'
     $uploadToCodecov = $true
   }
+
   # These are native MinGW builds, but they use the toolchain inside
   # MSYS2, this allows using all the dependencies and tools available
   # in MSYS2, but we cannot build inside the MSYS2 shell.
   $cmakeGenerator = 'MinGW Makefiles'
-  $cmakeGeneratorArgs = 'VERBOSE=1'
-
-  # Add MinGW to the PATH
-  $env:PATH = "C:\msys64\mingw$bits\bin;$env:PATH"
+  $cmakeGeneratorArgs = @('VERBOSE=1')
+  $env:CC = 'gcc'
+
+  @{
+    CMAKE_SH = 'CMAKE_SH-NOTFOUND'
+    CMAKE_MAKE_PROGRAM = "C:/msys64/mingw$bits/bin/mingw32-make.exe"
+  }.GetEnumerator() | ForEach-Object {
+    $nvimCmakeVars[$_.Key] = $_.Value
+    $depsCmakeVars[$_.Key] = $_.Value
+    $cmakeGeneratorArgs += $_.Key + '=' + $_.Value
+  }
 
   # Build third-party dependencies
   C:\msys64\usr\bin\bash -lc "pacman --verbose --noconfirm -Su" ; exitIfFailed
-  C:\msys64\usr\bin\bash -lc "pacman --verbose --noconfirm --needed -S mingw-w64-$arch-cmake mingw-w64-$arch-perl mingw-w64-$arch-diffutils mingw-w64-$arch-unibilium" ; exitIfFailed
+  C:\msys64\usr\bin\bash -lc "pacman --verbose --noconfirm --needed -S mingw-w64-$arch-gcc mingw-w64-$arch-make mingw-w64-$arch-cmake mingw-w64-$arch-perl mingw-w64-$arch-diffutils mingw-w64-$arch-unibilium" ; exitIfFailed
 }
 elseif ($compiler -eq 'MSVC') {
   $cmakeGeneratorArgs = '/verbosity:normal'
-  if ($bits -eq 32) {
-    $cmakeGenerator = 'Visual Studio 15 2017'
-  }
-  elseif ($bits -eq 64) {
-    $cmakeGenerator = 'Visual Studio 15 2017 Win64'
+  $cmakeGenerator = 'Visual Studio 15 2017'
+  if ($bits -eq 64) {
+    $cmakeGenerator += ' Win64'
   }
 }
 
-# Remove Git Unix utilities from the PATH
-$env:PATH = $env:PATH.Replace('C:\Program Files\Git\usr\bin', '')
-
 # Setup python (use AppVeyor system python)
-C:\Python27\python.exe -m pip install neovim ; exitIfFailed
-C:\Python35\python.exe -m pip install neovim ; exitIfFailed
-# Disambiguate python3
-move c:\Python35\python.exe c:\Python35\python3.exe
-$env:PATH = "C:\Python35;C:\Python27;$env:PATH"
-# Sanity check
-python  -c "import neovim; print(str(neovim))" ; exitIfFailed
-python3 -c "import neovim; print(str(neovim))" ; exitIfFailed
-
-$env:PATH = "C:\Ruby24\bin;$env:PATH"
-cmd /c gem.cmd install neovim ; exitIfFailed
-where.exe neovim-ruby-host.bat ; exitIfFailed
-
-cmd /c npm.cmd install -g neovim ; exitIfFailed
-where.exe neovim-node-host.cmd ; exitIfFailed
+# Perform sanity checks after installation
+if (Test-Path -PathType Container C:\Python27) {
+  $env:PATH = "C:\Python27;$env:PATH"
+}
+if (where.exe python.exe 2> $null) {
+  python.exe  -m pip install neovim ; exitIfFailed
```

Comment:
the "Disambiguate python3" step should be done before calling python. Otherwise it might work by accident, but that is too subtle, better to be explicit.